pyproject.toml

```diff
@@ -3,22 +3,21 @@ line-length = 120
 
 [tool.ruff.lint]
 ignore = ["E501", "G004", "PLR2004", "RUF005", "RUF012", "UP040"] # longlines, fstrings in logs, magic values, consider not concat, mutable classbits, type instead of TypeAlias
-extend-select = ["W", "I", "SLF", "SIM", "B", "Q", "A", "COM", "C90", "UP", "YTT", "ASYNC", "BLE", "C4", "DTZ", "T10", "DJ", "EXE" ]
-# extend-select = [ "S", "EM", "FA",
-#                  "ISC", "ICN", "G", "INP", "PIE", "T20", "PYI", "PT", "RSE", "RET", "SLOT", "TID", "TCH", "INT", "PTH",
-#                  "FIX", "PGH", "PL", "TRY", "FLY", "PERF", "RUF"]
+extend-select = ["G", "W", "I", "SLF", "SIM", "B", "Q", "A", "COM", "C90", "UP", "YTT", "ASYNC", "BLE", "C4", "DTZ", "T10", "DJ", "EXE", "S"]
+# extend-select = ["EM", "FA", "ISC", "ICN", "INP", "PIE", "T20", "PYI", "PT", "RSE", "RET",
+#                  "SLOT", "TID", "TCH", "INT", "PTH", "FIX", "PGH", "PL", "TRY", "FLY", "PERF", "RUF"]
 unfixable = ["ERA"]
 
 [tool.ruff.lint.extend-per-file-ignores]
-"tests/*" = ["S101"]  # `assert` is fine in tests
+"*" = ["S108"] # insecure hardcoded temp file name -- fix at some point?
```

Comment:
did we have a lot of these causing issues?

The fix should be something like:

```
with tempfile.TemporaryFile() as tmp:
    # Do stuff with tmp
    tmp.write('stuff')

```

Reply:
Just read the source code. Does seem a bit trickier due to the way we are doing things. Happy to leave for this PR since we run this in lambda but worth us fixing soon